maps/biter_battles_v2/ai.lua

```diff
@@ -256,15 +256,18 @@ end
 ai.main_attack = function()
 	local surface = game.surfaces["biter_battles"]
 	local force_name = global.next_attack
-	local biter_force_name = force_name .. "_biters"
-	local wave_amount = math.ceil(get_threat_ratio(biter_force_name) * 7)
 	
-	set_biter_raffle_table(surface, biter_force_name)
-	
-	for c = 1, wave_amount, 1 do		
-		create_attack_group(surface, force_name, biter_force_name)
+	if not global.training_mode or (global.training_mode and #game.forces[force_name].connected_players > 0) then
```

Comment:
this IF wrapper being added around the existing code is the actual change. Github is just being special showing so many lines as being modified.